pkg/internal/cluster/kubeconfig/kubeconfig.go

```diff
@@ -23,7 +25,9 @@ import (
 	"sigs.k8s.io/kind/pkg/errors"
 	"sigs.k8s.io/kind/pkg/internal/cluster/context"
 
-	"sigs.k8s.io/kind/pkg/internal/util/kubeconfig"
+	// this package has slightly more generic kubeconfig helpers
+	// and minimal dependencies on the rest of kind
+	"sigs.k8s.io/kind/pkg/internal/cluster/kubeconfig/internal/kubeconfig"
```

Comment:
too many internals?

Reply:
not at all 🙃

Reply:
the way go handles this is that for any given directory tree with `internal` as a segment, only packages rooted in the same directory can import it.

IE the package here can only be imported by packages with the prefix `sigs.k8s.io/kind/pkg/internal/cluster/kubeconfig/`

Reply:
this allows us to split up the code here into a tiering of unexported methods (the internal package, vs the unexported helpers) which is nicer than one big flat space of all lowercase helper methods.